Sender/Sender_Test.cpp

```diff
@@ -0,0 +1,19 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "test/catch.hpp"
+#include "Sender_Header.h"
+
+TEST_CASE("To check if file was successfully opened to read") 
```

Comment:
No tests for the output format? This is the end responsibility of the Sender.
The absence of tests here means that future mistakes in the output will only be caught when integrating with the receiver. That will take a lot more time than a simple unit-test